soh/soh/Enhancements/game-interactor/GameInteractionEffect.h

```diff
@@ -217,7 +217,7 @@ namespace GameInteractionEffect {
         void _Apply() override;
     };
 
-    class PressRandomButton: public GameInteractionEffectBase, public ParameterizedGameInteractionEffect {
+    class PressRandomButton: public GameInteractionEffectBase, public RemovableGameInteractionEffect {
         GameInteractionEffectQueryResult CanBeApplied() override;
         void _Apply() override;
     };
```

Comment:
Seems you could replace `GameInteractionEffectBase` with `RemovableGameInteractionEffect` and continue using `ParameterizedGameInteractionEffect` and having the `30` value set like it originally was.

This might also resolve the build error.

```suggestion
    class PressRandomButton: public RemovableGameInteractionEffect, public RemovableGameInteractionEffect {
        GameInteractionEffectQueryResult CanBeApplied() override;
        void _Apply() override;
    };
```

Reply:
Oh Huh I could've sworn it built locally. I'll try out your suggestion, that sounds much simpler. I'm a little lost in all the advanced C++ salsa sometimes.